### DISCUSSION TITLE
calculate the rule

### DISCUSSION THREAD

**Discussion by Cheney-77** [jertel/elastalert2#1272] [Q&A] [answered]
hello sir,my es doc like this: 
{
  "_index": "metricbeat-7.17.12-2023.09.14-000001",
  "_type": "_doc",
  "_id": "Z69ztooB8jhY0hacU117",
  "_version": 1,
  "_score": 1,
  "_source": {
    "@timestamp": "2023-09-21T06:35:25.071Z",
    "service": {
      "type": "system"
    },
    "system": {
      "cpu": {
        "system": {
          "pct": 0.0832,
          "norm": {
            "pct": 0.0104
          }
        },
        "iowait": {
          "norm": {
            "pct": 0.0025
          },
          "pct": 0.0201
        },
        "total": {
          "pct": 0.2648,
          "norm": {
            "pct": 0.0331
          }
        },
        "nice": {
          "pct": 0,
          "norm": {
            "pct": 0
          }
        },
        "irq": {
          "norm": {
            "pct": 0
          },
          "pct": 0
        },
        "steal": {
          "pct": 0.001,
          "norm": {
            "pct": 0.0001
          }
        },
        "cores": 8,
        "user": {
          "pct": 0.1725,
          "norm": {
            "pct": 0.0216
          }
        },
        "softirq": {
          "norm": {
            "pct": 0.001
          },
          "pct": 0.008
        },
        "idle": {
          "pct": 7.7151,
          "norm": {
            "pct": 0.9644
          }
        }
      }
    },
    "ecs": {
      "version": "1.12.0"
    },
    "agent": {
      "hostname": "hanyun",
      "ephemeral_id": "57b267f7-9a7c-4242-9cfa-a3bd2bd18a4e",
      "id": "0aac6045-bc07-4a6e-b644-5f19e1487d0b",
      "name": "10.90.20.207",
      "type": "metricbeat",
      "version": "7.17.12"
    },
    "host": {
      "cpu": {
        "usage": 0.0331
      },
      "name": "10.90.20.207"
    },
    "event": {
      "dataset": "system.cpu",
      "module": "system",
      "duration": 117489
    },
    "metricset": {
      "name": "cpu",
      "period": 10000
    }
  },
  "fields": {
    "system.cpu.iowait.norm.pct": [
      0.003
    ],
    "system.cpu.system.pct": [
      0.083
    ],
    "system.cpu.nice.norm.pct": [
      0
    ],
    "system.cpu.total.pct": [
      0.265
    ],
    "system.cpu.irq.pct": [
      0
    ],
    "system.cpu.irq.norm.pct": [
      0
    ],
    "system.cpu.user.norm.pct": [
      0.022
    ],
    "system.cpu.total.norm.pct": [
      0.033
    ],
    "service.type": [
      "system"
    ],
    "system.cpu.cores": [
      8
    ],
    "agent.type": [
      "metricbeat"
    ],
    "system.cpu.iowait.pct": [
      0.02
    ],
    "event.module": [
      "system"
    ],
    "system.cpu.softirq.pct": [
      0.008
    ],
    "system.cpu.idle.norm.pct": [
      0.964
    ],
    "agent.name": [
      "10.90.20.207"
    ],
    "host.name": [
      "10.90.20.207"
    ],
    "system.cpu.idle.pct": [
      7.715
    ],
    "system.cpu.softirq.norm.pct": [
      0.001
    ],
    "system.cpu.nice.pct": [
      0
    ],
    "metricset.period": [
      10000
    ],
    "system.cpu.steal.pct": [
      0.001
    ],
    "system.cpu.steal.norm.pct": [
      0
    ],
    "agent.hostname": [
      "hanyun"
    ],
    "system.cpu.system.norm.pct": [
      0.01
    ],
    "metricset.name": [
      "cpu"
    ],
    "event.duration": [
      117489
    ],
    "@timestamp": [
      "2023-09-21T06:35:25.071Z"
    ],
    "agent.id": [
      "0aac6045-bc07-4a6e-b644-5f19e1487d0b"
    ],
    "ecs.version": [
      "1.12.0"
    ],
    "host.cpu.usage": [
      0.033
    ],
    "system.cpu.user.pct": [
      0.173
    ],
    "agent.ephemeral_id": [
      "57b267f7-9a7c-4242-9cfa-a3bd2bd18a4e"
    ],
    "agent.version": [
      "7.17.12"
    ],
    "event.dataset": [
      "system.cpu"
    ]
  }
}

if i want to use (system.cpu.user.pct+system.cpu.system.pct)/system.cpu.cores > 0.8 to trigger the alarm,how can i achieve it? tkx@jertel

**Comment by jertel** [comment, marked as answer]
You can either use an aggregation script, or you can do the math in an ingest pipeline.
